python/plugins/processing/algs/gdal/GdalAlgorithm.py

```diff
@@ -89,6 +89,7 @@ def getOgrCompatibleSource(self, parameter_name, parameters, context, feedback,
                 # and extract selection if required
                 ogr_data_path = self.parameterAsCompatibleSourceLayerPath(parameters, parameter_name, context,
                                                                           QgsVectorFileWriter.supportedFormatExtensions(),
+                                                                          'gpkg',
```

Comment:
Please, don't hardcode this format (even if it's better than shp :) )

`QgsVectorFileWriter.supportedFormatExtensions()[0]` returns 'gpkg' is GDAL have gpkg support (it can be disabled) instead 'shp'.

Reply:
I wouldn't call it hardcoded... but who am I? :-p

If gdal has support to gpkg, why isn't the first format from that list used? It must be forced somewhere else. No?